v4/internal/otel_expect.go

```diff
@@ -6,14 +6,66 @@ import (
 
 // OpenTelemetryExpect implements internal.Expect for use in testing.
 type OpenTelemetryExpect struct {
-	*testtrace.Tracer
+	Spans *testtrace.StandardSpanRecorder
 }
 
-// ExpectTxnEvents TODO
-func (e *OpenTelemetryExpect) ExpectTxnEvents(t Validator, want []WantEvent) {}
+func expectSpan(t Validator, want WantSpan, span *testtrace.Span) {
+	t.Helper()
+	if want.Name != MatchAnyID {
+		if name := span.Name(); name != want.Name {
+			t.Errorf("Incorrect span name:\n\texpect=%s actual=%s",
+				want.Name, name)
+		}
+	}
+	spanCtx := span.SpanContext()
+	if want.SpanID != MatchAnyID {
+		if id := spanCtx.SpanID.String(); id != want.SpanID {
+			t.Errorf("Incorrect span id:\n\texpect=%s actual=%s",
+				want.SpanID, id)
+		}
+	}
+	if want.TraceID != MatchAnyID {
+		if id := spanCtx.TraceID.String(); id != want.TraceID {
+			t.Errorf("Incorrect trace id:\n\texpect=%s actual=%s",
+				want.TraceID, id)
+		}
+	}
+	if want.ParentID != MatchAnyID {
+		id := span.ParentSpanID().String()
+		noParent := "0000000000000000"
+		if want.ParentID == MatchNoParent {
+			if id != noParent {
+				t.Errorf("Incorrect parent id:\n\texpect=%s actual=%s",
+					noParent, id)
+			}
+		} else if want.ParentID == MatchAnyParent {
+			if id == noParent {
+				t.Error("Incorrect parent id: expected a parent but found none")
+			}
+		} else if id != want.ParentID {
+			t.Errorf("Incorrect parent id:\n\texpect=%s actual=%s",
+				want.ParentID, id)
+		}
```

Comment:
For the `ParentID` field I implemented two additional options for the cases where the span has no parent and the case where the span has a parent but we don't care what the specific parent ID is. Thoughts here welcome as well.